### PR TITLE
Add core governance document

### PR DESCRIPTION
**Pull request by TheCurle**
Defines the project, our general processes, and the teams under our jurisdiction.

### CODE REVIEW

File: core.md
Comment (Matyrobbrt) [inline core.md]:
```suggestion
The NeoForged project is split between [GitHub](https://github.com/NeoForged) and [Discord](https://discord.neoforged.net). It is a community of volunteers, maintaining a free and open source ecosystem of modifications to the Minecraft game.
```

File: core.md
Comment (Matyrobbrt) [inline core.md]: Direct doesn't seem like a good fit for the sentence. Maybe steer?

File: core.md
Comment (Matyrobbrt) [inline core.md:24]: Are you sure it's *not* meant to *prevent* this phenomenon?

File: core.md
Comment (Matyrobbrt) [inline core.md]:
```suggestion
The Code of Conduct of the project is taken to be the content of [the Contributor Covenant](https://www.contributor-covenant.org/version/2/1/code_of_conduct/), so this document is not the Code of Conduct.
```
We should still probably have our own CoC

File: core.md
Comment (Matyrobbrt) [inline core.md]: If you capitalize Repository then you should do so for Subproject as well.

File: core.md
Comment (Matyrobbrt) [inline core.md]: Considering the lack of punctuation these sentences should start with a lowercase letter and end in a semicolon.
Also, NeoForge is not a mod loader, it is a framework, and the framework needs to be updated, not FML.
Capitalize `repository` in `multi-repository` too.

File: core.md
Comment (Matyrobbrt) [inline core.md]: The short URL should be renamed, and it should be under our domain (as is the Discord invite) so it can be updated later on if needs be. Additionally, which are those "some" teams (i.e. document the requirements in their specific subsections).

File: core.md
Comment (Matyrobbrt) [inline core.md]: Council is a singular noun.

File: core.md
Comment (Matyrobbrt) [inline core.md]: The way this is worded would imply that the Council as a team has opinions. I would say it "benefits from fresh opinions"

File: core.md
Comment (Matyrobbrt) [inline core.md:56]: This section should mention that the powers of the council are not all-encompassing and that decisions on different matters shall be first ran by the team responsible and may only be vetoed by the council under exceptional reasoned circumstances (“drama”, draws).

File: core.md
Comment (Matyrobbrt) [inline core.md]: Again, decide on repository or Repository.

File: core.md
Comment (Matyrobbrt) [inline core.md]:
```suggestion
Though Maintainers may individually or collectively have different access rights across the Project's many Repositories or Infrastructure services, all Maintainers within the team are considered equal to each other with respect to their rights within the team's governance.
```

File: core.md
Comment (Matyrobbrt) [inline core.md]: "above" being? (link)

File: core.md
Comment (Matyrobbrt) [inline core.md]: Full stop at the end of each sentence or lowercase and semicolons.

Comment (Matyrobbrt) [inline core.md]: Why the Council directly and not the team assigned to the project first?

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: core.md
Comment (TheCurle) [reply core.md]: It's about project direction, so the verb "to direct" seems the best fit here.

File: core.md
Comment (TheCurle) [reply core.md:24]: Yes. The point of the document is to give guidance, not to have such excruciating detail that it'll prevent people trying to rules lawyer to governance.

It is not intended to prevent rules lawyering.

File: core.md
Comment (TheCurle) [reply core.md]: If there's a pressing need over the Contributor Covenant, I can see that, yeah.

I'm not sure if there's anything we need that isn't covered, to be honest. Nobody is above the rules.

File: core.md
Comment (TheCurle) [reply core.md]: Right, yeah, i forgot that we split the loader out from the framework.

File: core.md
Comment (TheCurle) [reply core.md]: Good call.

File: core.md
Comment (TheCurle) [reply core.md]: The people inside it are not, though; good catch nonetheless.

File: core.md
Comment (TheCurle) [reply core.md]: Fresh insight from a new team member, a different view on situations. Yes, good call.

File: core.md
Comment (TheCurle) [reply core.md]: I specifically meant major architectural decisions about the overall Project / framework, outside of the exclusive jurisdiction of the subproject.

File: core.md
Comment (marchermans) [inline core.md]: Maybe indicate that we are free to extend or overrule the covenant if we ever deem it to be needed. People can get annoyed if we do not clarify that ahead of time.

File: core.md
Comment (marchermans) [inline core.md]: Conflict resolution is not the responsibility of the moderation team IMHO.
They are the community managers and moderators but have no direct moderation power within the organization, that should be the steering council only.

File: core.md
Comment (marchermans) [inline core.md:93]: Correct, but maybe stipulate that they do not influence technical decisions.

File: core.md
Comment (marchermans) [inline core.md:85]: Possibly state that they have no community moderation powers, and that the moderators are supposed to manage the community and its expectations.

File: core.md
Comment (marchermans) [inline core.md:125]: IMHO: We should add something like: `However, continuously putting them under question is not allowed. They should be able to do their job and should regularly check in with the rest of the maintainers to collect feedback.`

**Review by marchermans — CHANGES_REQUESTED** [review]

File: core.md
Comment (Matyrobbrt) [reply core.md:93]: This will lead to the ”but what if a moderator is a maintainer?” question. I don't find such statement needed given the implications.

File: core.md
Comment (Matyrobbrt) [inline core.md]: Use smart quotes (`“”`)

File: core.md
Comment (TheCurle) [reply core.md]: Resolving conflicts between community members is what I was going for, here. That's the job of the Moderators. It's part of the definition of "moderating" a discussion.

Comment (marchermans) [reply core.md]: Specify that it is "Conflict resolution with the community is the responsibility of the moderation team, while conflict resolution within the organization is the responsibility of the steering council". This prevents any misunderstandings about the relevant roles.

**Review by marchermans — DISMISSED** [review]

File: core.md
Comment (Matyrobbrt) [inline core.md:36]:
```suggestion
"The Project" refers to The NeoForged project, the organisation, with a presence on both GitHub and Discord, as noted at the start of this document. It does not refer to any specific repository or subproject.
```

**Review by Matyrobbrt — DISMISSED** [review]

File: core.md
Comment (TheCurle) [reply core.md]: I have absolutely no idea how.

Comment (Matyrobbrt) [reply core.md]:
```suggestion
They “steer” the project to keep it aligned with the long-term vision, hence its secondary name of Vision Team.
```

**Review by Technici4n — APPROVED** [review]

File: core.md
Comment (ChampionAsh5357) [inline core.md:94]: How are moderators appointed? Do they also have the same process as maintainers? Or are they slightly different?

Comment (TheCurle) [reply core.md:94]: Good point! They share the same application system.

File: core.md
Comment (Matyrobbrt) [inline core.md]: This wording sounds as if their opinion won't be heard and simply dismissed. Maybe clarify that they're free to comment and their opinion will be considered just as of any other person's.

File: core.md
Comment (Matyrobbrt) [inline core.md]: What is "technical rights"?

File: core.md
Comment (Matyrobbrt) [inline core.md:47]: Consider clarifying that the maintainers and moderators team don't have to be odd as the council will break ties.

**Review by Matyrobbrt — DISMISSED** [review]

File: core.md
Comment (TheCurle) [reply core.md]: This one was taken from @sciwhiz12's draft, you'd have to ask him.

Comment (sciwhiz12) [reply core.md]: Technical rights/access means the user has access to technical configuration or operations. It can take many forms, according to their context:
- For GitHub repositories, it chiefly refers to Write permission and above; having the permission to push commits to branches.
- For services, it refers to being able to reconfigure, enable, or disable service components.
- For server access, it refers to the access to login and access a console or configuration page for a server.

It is qualified as "technical" to distinguish it from other forms of rights/access, such as moderation rights/access (the ability to moderate users and their contents).

File: core.md
Comment (Technici4n) [reply core.md:47]: They should usually not be asked for a full vote anyway, so I don't think it matters.

**Review by Technici4n — APPROVED** [review]

**Review by marchermans — APPROVED** [review]

File: core.md
Comment (PaintNinja) [inline core.md:82]: Seemingly contradictory language, assuming "their rights within the team's governance" refers to the definition above regarding it being those "who have push access or technical rights to the project's repositories or infrastructure."

File: core.md
Comment (PaintNinja) [inline core.md:26]: I am against the adoption of the contributor covenant due to its usage of politically contentious language and associated concerning FAQ, but can be on board with an altered version.

**Review by PaintNinja — CHANGES_REQUESTED** [review]

**Review by SizableShrimp — APPROVED** [review]

File: core.md
Comment (Technici4n) [reply core.md:26]: What are you referring to in particular?

File: core.md
Comment (cpw) [reply core.md:82]: I don't see any contradiction here? You're going to have to be super extremely explicit and outline precisely what you mean.

File: core.md
Comment (cpw) [reply core.md:26]: I disagree. Contributors covenant is a widely accepted document for operating within public online spheres. It's only "political" controversy is from those who wish not to be bound by some of it's terms.

**Review by 0xilly — APPROVED** [review]

**Review by AterAnimAvis — APPROVED** [review]

**Comment by TelepathicGrunt** [comment]
Is there any timeframe of when this can be merged? Is there anything that is actually holding this up?

**Review by TelepathicGrunt — APPROVED** [review]

File: core.md
Comment (PaintNinja) [reply core.md:82]: > Maintainers may individually or collectively have different access rights

and

> Maintainers within the team are considered equal to each other with respect to their rights

within the same sentence.

Comment (TheCurle) [reply core.md:82]: Access rights (permission to push to different repositories) and rights in the team (whose opinion is weighted more) are taken to be separate subjects entirely.

Comment (sciwhiz12) [reply core.md:82]: You left out a key part of that second sentence (emphasis mine):

> _[...]_ all Maintainers within the team are considered equal to each other with respect to their rights **within the team's governance.**

That is, while Maintainers hold different levels of access to resources, repositories, or infrastruture, they have the same 'loudness of voice' and rights (e.g. voting rights) with each other in the Maintainers team.

Comment (PaintNinja) [reply core.md:82]: Shouldn't those who specialise in subprojects (such as subproject leaders) be given additional considerations based on their knowledge of a given area, rather than treating a new maintainer and an experienced specialist equally?

Comment (TheCurle) [reply core.md:82]: Yes. This is why there's a separate topic for Subproject Leaders explicitly outlining their authority and rights exceeding standard maintainers.

Comment (cpw) [reply core.md:82]: OK. this looks to be resolved to me. Paint wasn't understanding the context of access rights to repositories vs rights within the organization.

**Pull request merged**
